rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -22,31 +23,51 @@ const DUMMY_KEY: &str = "dummy";
 /// Converts a connection struct into a HashMap that can be sent over D-Bus.
 ///
 /// * `conn`: Connection to convert.
-pub fn connection_to_dbus(conn: &Connection) -> NestedHash {
+pub fn connection_to_dbus<'a>(
+    conn: &'a Connection,
+    controller: Option<&'a Connection>,
+) -> NestedHash<'a> {
     let mut result = NestedHash::new();
-    let mut connection_dbus = HashMap::from([
-        ("id", conn.id().into()),
-        ("type", ETHERNET_KEY.into()),
-        ("interface-name", conn.interface().into()),
-    ]);
+    let mut connection_dbus =
+        HashMap::from([("id", conn.id().into()), ("type", ETHERNET_KEY.into())]);
+
+    if let Some(interface) = &conn.interface() {
+        connection_dbus.insert("interface-name", interface.to_owned().into());
+    }
+
+    if let Some(controller) = controller {
+        connection_dbus.insert("slave-type", "bond".into()); // TODO: only 'bond' is supported
+        let master = controller.interface().unwrap_or(controller.id());
+        connection_dbus.insert("master", master.into());
+    }
+
     result.insert("ipv4", ip_config_to_ipv4_dbus(conn.ip_config()));
     result.insert("ipv6", ip_config_to_ipv6_dbus(conn.ip_config()));
     result.insert("match", match_config_to_dbus(conn.match_config()));
 
-    if conn.is_ethernet() {
-        let ethernet_config =
-            HashMap::from([("assigned-mac-address", Value::new(conn.mac_address()))]);
-        result.insert(ETHERNET_KEY, ethernet_config);
-    } else if let Connection::Wireless(wireless) = conn {
-        connection_dbus.insert("type", WIRELESS_KEY.into());
-        let wireless_dbus = wireless_config_to_dbus(wireless);
-        for (k, v) in wireless_dbus {
-            result.insert(k, v);
+    match &conn {
+        Connection::Ethernet(_) | Connection::Loopback(_) => {
```

Comment:
I think `is_ethernet()` was the hook to identify connections which need to set the customized macaddr via `802-3-ethernet.assigned-mac-address`. AFAIK Dummy was one of them and Bond should be one.

If we do it this way. We would need to set mac in the corresponding match block of `Connection::Bond` and `Connection::Dummy`.
Or we extract the writing of `802-3-ethernet.assigned-mac-address` out of this match block.